update(UI overview): add bonita apps (#1362)
bonitasoft/bonita-doc: md/design-methodology.md

* People: Who owns the application? Who owns the processes in the application? Who uses the application and processes? Who are the key
users? Who just needs to know the application exists? When and how is it used? Is it used frequently, on a schedule, in emergencies? Is
it used by just one person or by many people at the same time?
* Information: what information does it deliver? What are the expected results of the processes, and of the application?  
There are three main kinds of applications based on the information they deliver, and the choice made may impact the processes:
  * Tasklist applications: all processes mixed, users do tasks by priority. Bonita Portal and Bonita Applications deliver such a user experience. You can use them as they are, or customize Bonita Applications to make them closer to the users' needs. We will see how below.

>   * Business centered-applications. The processes help managing the lifecycle of business objects instances, like a request. Users needs to act on specific requests at a given time, and to do so, they need to filter the requests list and have actions linked to any request to make its status evolve. These are business-specific and need to be developed as Living Applications.  

Reviewer: ```suggestion
  * Business centered-applications. The processes help managing the lifecycle of business objects instances, like a request. Users needs to act on specific requests at a given time, and to do so, they need to filter the requests list and have actions linked to the requests to make their status evolve. These are business-specific and need to be developed as Living Applications.  
```
just a suggestion, but maybe I am not getting the exact meaning ?

* Process maintainability. Add annotations to explain the process. Make sure all elements have descriptive labels. Generate the process document and check it.

## Designing the application 

When the application definition and process definitions are complete, you will have defined what interaction users have with application pages and with each process task. Create the pages and forms needed. 

### From a Bonita application
Each application is devoted to a special business need:
1. Bonita Administrator Application for the management, monitoring and operations of Bonita Runtime 
1. Bonita User Application for task execution
1. Adaptive Case Management example for Case Management. Note that this example is only available for the Enterprise edition.

To customize an application to your needs:
1. From Bonita Studio's Welcome page, in the "Resources" tile, download the application that suits your needs. 
In the studio "Project explorer" pane, you will find all resources of the application: descriptor, layout, theme, custom widgets, as well as some pages.

> Indeed, our pages have been developed with different frameworks over time. Pages developed with the UI Designer appear in the Project explorer, they can be edited in the UI Designer; pages developed with AngularJS, React, or PortalJS are embedded in the Development Suite Tomcat and can be found in the Resources page of Bonita Portal / the Bonita Administrator Application. Those pages have been "wrapped" from Bonita Portal to be used as custom pages. They cannot be edited in the UI Designer.

Reviewer: ```suggestion
Our pages have been developed with different frameworks over time.
- Pages developed with the UI Designer appear in the Project explorer, they can be edited in the UI Designer
- Pages developed with AngularJS are embedded in the Development Suite Tomcat and can be found in the Resources page of Bonita Portal / the Bonita Administrator Application. Those pages have been "wrapped" from Bonita Portal to be used as custom pages. They cannot be edited in the UI Designer.
- Pages developed with React are embedded in the Development Suite Tomcat and can be found in the Resources page of Bonita Portal / the Bonita Administrator Application. Those pages cannot be edited in the UI Designer but they can be modified using an IDE (Integrated Development Environment) if needed.
```

2. Duplicate the application descriptor and rename your copy
3. Delete the pages you do not need, add the pages you need, keep the pages that fit the target use  
4. You can customize the pages created with the UI Designer. To do so:
    1. From the project explorer pane, double-click on the page to open it in the UI Designer
    2. Make it yours
    3. We recommend that you save it under another name, in case you need the original version at some point

::: info
Note: While both the Bonita User Application and the Bonita Administrator Application are supported by Bonitasoft in their original versions, any customization of yours cannot be supported, as Bonitasoft cannot garantee the content of your custom development.  
Similarly, the Adaptive Case Management example, which can only be used after it is customized, is not supported.
However, our team of Professional Services consultants can assist you in the customization phase.
:::

#### List of Bonita Applications customizable pages

##### Bonita User Application
In this application, you can customize:

>   * Bonita User Case List `UserCaseListBonitaV3`

Reviewer: we should avoid putting the version as is is updated regularly and we won't update this page when we do so.
I propose V_

Author: ok
